### PR TITLE
Remove Python 2.7 support

### PR DESCRIPTION
**Pull request by dkfellows**
PyPI and pip are forcing the issue; we can't keep going much longer.
SpiNNakerManchester/SpiNNUtils#132

### CODE REVIEW

**Comment by dkfellows** [comment]
Should I move `require_subclass` into SpiNNUtilities?

**Comment by Christian-B** [comment]
> Should I move `require_subclass` into SpiNNUtilities?

Where is this code?

**Comment by dkfellows** [comment]
Moved.

File: spinn_front_end_common/abstract_models/abstract_provides_incoming_partition_constraints.py [hunk 1]
```diff
@@ -13,12 +13,14 @@
...
 
-from six import add_metaclass
 from spinn_utilities.abstract_base import AbstractBase, abstractmethod
+from spinn_utilities.require_subclass import require_subclass
+from pacman.model.graphs import AbstractVertex
 
 
-@add_metaclass(AbstractBase)
-class AbstractProvidesIncomingPartitionConstraints(object):
+@require_subclass(AbstractVertex)
```
Comment (alan-stokes) [inline spinn_front_end_common/abstract_models/abstract_provides_incoming_partition_constraints.py:21]: might want to chekc the interface function for using those. i think its only machine level these days

Comment (alan-stokes) [reply spinn_front_end_common/abstract_models/abstract_provides_incoming_partition_constraints.py:21]: acutally, seems we still support both. maybe we should only support it on 1 though? food for thought

File: spinn_front_end_common/utility_models/data_speed_up_packet_gatherer_machine_vertex.py [hunk 2]
```diff
@@ -669,12 +666,8 @@ def send_data_into_spinnaker(
             original_data = bytes(data[offset:n_bytes + offset])
             verified_data = bytes(transceiver.read_memory(
                 x, y, base_address, n_bytes))
-            if PY2:
-                self.__verify_sent_data_py2(
-                    original_data, verified_data, x, y, base_address, n_bytes)
-            else:
-                self.__verify_sent_data_py3(
-                    original_data, verified_data, x, y, base_address, n_bytes)
+            self.__verify_sent_data_py3(
```
Comment (alan-stokes) [inline spinn_front_end_common/utility_models/data_speed_up_packet_gatherer_machine_vertex.py:669]: maybe remove the py3 these days?

**Pull request merged**